spec/unit/http/client_spec.rb

```diff
@@ -569,16 +570,27 @@ def redirect_to(status: 302, url:)
       expect(response).to be_success
     end
 
-    it "preserves query parameters given a relative location" do
+    it "applies query parameters from the location header" do
       relative_url = "/people.html"
       query = { 'debug' => true }
-      stub_request(:get, start_url).with(query: query).to_return(redirect_to(url: relative_url))
-      stub_request(:get, "https://www.example.com:8140/people.html").with(query: query).to_return(status: 200)
+      stub_request(:get, start_url).with(query: query).to_return(redirect_to(url: "#{relative_url}?redirected=true"))
+      stub_request(:get, "https://www.example.com:8140/people.html").with(query: {'redirected' => 'true'}).to_return(status: 200)
```

Comment:
I think this test would read better if the `query` variable contents were to be replaced with `{ 'redirected' => true }`, and then interpolated on line 577, since right now the content of `query` can be anything and doesn't really influence the test.

Reply:
Ah, for this test I wanted to make sure puppet uses the query from the location header (`'redirected'=> true`), not the query that it originally started with (`'debug' => true`). Maybe it would be better to do:

```
stub_request(:get, start_url).with(query: {'redirected' => false}).to_return(redirect_to(url: "#{relative_url}?redirected=true"))
```
and then assert that that the second request has `'redirected' => true`?

Reply:
Yeah, I think that would make more sense!

Reply:
Cool, updated